File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        ArrayList<Integer> answer = new ArrayList<>();
        int oldValue = -1;
        
        for (int num : arr) {
            if (num != oldValue) {
                answer.add(num);
                oldValue = num;
            }
        }
        
        return answer.stream()
            .mapToInt(Integer::intValue)
            .toArray();
    }
}

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.cpp
#include <vector>
#include <iostream>

using namespace std;

vector<int> solution(vector<int> arr) 
{
    vector<int> answer;

    int oldValue = -1;
    for (int num : arr) {
        if (num != oldValue) {
            answer.push_back(num);
            oldValue = num;
        }
    }

    return answer;
}